### PR TITLE
DOC-11078: Creates Scopes and Collections Config page and child page in the nav

### PR DESCRIPTION
**Pull request by franrob-projects**
- Creates Scopes and Collections Config page 
- Child page in the nav

### CODE REVIEW

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 1]
```diff
@@ -0,0 +1,191 @@
+= Scopes and Collections Configuration for Sync Gateway
+:page-aliases: learn/scopes-and-collections-config.adoc
+ifdef::show_edition[:page-edition: {release}]
+ifdef::prerelease[:page-status: {prerelease}]
+:page-role:
+:page-content: conceptual
+// :description: pass:q[About Sync Gateway _Roles_ and their part in secure cloud-to-edge enterprise data synchronization.]
```
Comment (dgnicholson) [inline modules/ROOT/pages/scopes-and-collections-config.adoc]: Remove comment?

```suggestion
:description: pass:q[How to configure Scopes and Collections for Sync Gateway - Examples with descriptions.]
```

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+Couchbase encourage you to explore and experiment with this concept within your Couchbase instance for practical experience.
+
+== Configuration Options
+
+You have these options to configure Scopes and Collections Sync for Gateway:
+
+* *custom scope/custom collection (the option we cover here)* +
+Gain maximum flexibility in organizing and accessing your data by customizing it to suit your needs.
+
+[Note]
```
Comment (dgnicholson) [inline modules/ROOT/pages/scopes-and-collections-config.adoc]: Needs to be uppercase.

```suggestion
[NOTE]
```

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+If it is not specified in the configuration, the previous versions are still compatible with the new version. Therefore, the provided examples are not necessary for upgrading to the latest version of Sync Gateway.
+====
+
+* *default scope/custom collection* +
+Create a custom collection with the default scope can lead to efficient data organization and quicker access control management.
+
+* *default scope/default collection* +
+Work within the default Scopes and Collections. This makes managing data easy and simple without creating extra scopes or collections.
+
+[Note]
```
Comment (dgnicholson) [inline modules/ROOT/pages/scopes-and-collections-config.adoc]:
```suggestion
[NOTE]
```

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+----
+function(doc, oldDoc, meta) {
+    channel("collection1");
+}
+----
+
+Whenever a document is synced or updated in `collection2`, its sync function is triggered. 
+This function calls the `channel` function with `collection1` as an argument, which means it will communicate with the sync function of `collection1`.
+
+[Note]
```
Comment (dgnicholson) [inline modules/ROOT/pages/scopes-and-collections-config.adoc]:
```suggestion
[NOTE]
```

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+----
+function(doc) {
+    return doc.type == "mobile";
+}
+----
+
+When importing documents for `collection2`, the import filter applies the same criteria as `collection1`. 
+This filter is based on the value of the `type` field and is used to filter the documents.
+
+[Note]
```
Comment (dgnicholson) [inline modules/ROOT/pages/scopes-and-collections-config.adoc]:
```suggestion
[NOTE]
```

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+ifdef::show_edition[:page-edition: {release}]
+ifdef::prerelease[:page-status: {prerelease}]
+:page-role:
+:page-content: conceptual
+// :description: pass:q[About Sync Gateway _Roles_ and their part in secure cloud-to-edge enterprise data synchronization.]
+
+For a successful configuration of your Couchbase Sync Gateway Scopes and Collections, use this helpful example. 
+First, it's important to remember that buckets can hold multiple scopes, and each scope can contain multiple collections.
+
+When you create a bucket, a bucket._default._default element is inherently included, as per the documentation on xref:server/learn/data/scopes-and-collections.adoc#default-scope-and-collection[Default Scopes and Collections] in Couchbase Server
```
Comment (dgnicholson) [inline modules/ROOT/pages/scopes-and-collections-config.adoc]: This is hard to parse without some snippet styling. Missing period.
```suggestion
When you create a bucket, a `bucket._default._default` element is inherently included, as per the documentation on xref:server/learn/data/scopes-and-collections.adoc#default-scope-and-collection[Default Scopes and Collections] in Couchbase Server.
```

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+:description: pass:q[How to configure Scopes and Collections for Sync Gateway - Examples with descriptions.]
+
+For a successful configuration of your Couchbase Sync Gateway Scopes and Collections, use this helpful example. 
+First, it's important to remember that buckets can hold multiple scopes, and each scope can contain multiple collections.
+
+When you create a bucket, a `bucket._default._default` element is inherently included, as per the documentation on xref:server/learn/data/scopes-and-collections.adoc#default-scope-and-collection[Default Scopes and Collections] in Couchbase Server.
+
+The default is unique to the bucket creation process and predominantly serves the purpose of backward compatibility.
+
+Please be aware that only Couchbase Server (CBS) allows using the word 'default' in naming conventions.
```
Comment (torcolvin) [inline modules/ROOT/pages/scopes-and-collections-config.adoc]:
```suggestion
Please be aware that only Couchbase Server (CBS) allows using the word `_default` in naming conventions. 
```

These naming conventions are private to CBS, not accessible to users.

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+
+For a successful configuration of your Couchbase Sync Gateway Scopes and Collections, use this helpful example. 
+First, it's important to remember that buckets can hold multiple scopes, and each scope can contain multiple collections.
+
+When you create a bucket, a `bucket._default._default` element is inherently included, as per the documentation on xref:server/learn/data/scopes-and-collections.adoc#default-scope-and-collection[Default Scopes and Collections] in Couchbase Server.
+
+The default is unique to the bucket creation process and predominantly serves the purpose of backward compatibility.
+
+Please be aware that only Couchbase Server (CBS) allows using the word 'default' in naming conventions. 
+This explains why you may see the term _default.default when creating a new bucket.
```
Comment (torcolvin) [inline modules/ROOT/pages/scopes-and-collections-config.adoc]:
```suggestion
This explains why you may see the term `_default._default` when creating a new bucket. 
```

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+== Configuration Options
+
+You have these options to configure Scopes and Collections Sync for Gateway:
+
+* *custom scope/custom collection (the option we cover here)* +
+Gain maximum flexibility in organizing and accessing your data by customizing it to suit your needs.
+
+[NOTE]
+====
+Scopes is not required for upgrading to Sync Gateway 3.1.
```
Comment (torcolvin) [inline modules/ROOT/pages/scopes-and-collections-config.adoc]:
```suggestion
The use of collections is not required for upgrading to Sync Gateway 3.1. 
```
We generally refer to things as "collections" even though all collections have a scopes parameter which would be the internal and config file name.

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+
+* *default scope/custom collection* +
+Create a custom collection with the default scope can lead to efficient data organization and quicker access control management.
+
+* *default scope/default collection* +
+Work within the default Scopes and Collections. This makes managing data easy and simple without creating extra scopes or collections.
+
+[NOTE]
+====
+Importing a dataset created in previous versions of Couchbase Server, the documents will now be saved automatically in the "_default" collection and scope.
```
Comment (torcolvin) [inline modules/ROOT/pages/scopes-and-collections-config.adoc:39]:
```suggestion
Importing a dataset created in previous versions of Couchbase Server, the documents will now be saved automatically in the `_default` scope and `_default` collection.
```

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+====
+
+== Configuration Example: _custom scope/_custom collection
```
Comment (torcolvin) [inline modules/ROOT/pages/scopes-and-collections-config.adoc]:
```suggestion
== Configuration Example: custom scope/custom collection
```

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+ifdef::show_edition[:page-edition: {release}]
+ifdef::prerelease[:page-status: {prerelease}]
+:page-role:
+:page-content: conceptual
+:description: pass:q[How to configure Scopes and Collections for Sync Gateway - Examples with descriptions.]
+
+For a successful configuration of your Couchbase Sync Gateway Scopes and Collections, use this helpful example. 
+First, it's important to remember that buckets can hold multiple scopes, and each scope can contain multiple collections.
+
+When you create a bucket, a `bucket._default._default` element is inherently included, as per the documentation on xref:server/learn/data/scopes-and-collections.adoc#default-scope-and-collection[Default Scopes and Collections] in Couchbase Server.
```
Comment (torcolvin) [inline modules/ROOT/pages/scopes-and-collections-config.adoc]: I don't think this needs to be named here, but this is referred to as a `keyspace` in N1QL https://docs.couchbase.com/server/current/n1ql/n1ql-intro/sysinfo.html

Keyspaces in sync gateway are similar except they are `<db>.<scope>.<collection>`

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 13]
```diff
@@ -0,0 +1,191 @@
...
+function(doc, oldDoc, meta) {
+    if (doc.type != "default") {
+        throw({forbidden : "Rejected document"});
+    }
+    channel("legacy");
+}
+----
+While the example sync function here has a specific purpose, it's crucial to understand that sync functions can perform various actions.
+
+Whenever a document is synced or updated in collection1, a function called `sync` is executed.
```
Comment (torcolvin) [inline modules/ROOT/pages/scopes-and-collections-config.adoc:129]: link to https://docs.couchbase.com/sync-gateway/current/sync-function.html ?

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 14]
```diff
@@ -0,0 +1,191 @@
...
+If the type field of the document is not `default`, the function will throw an error message saying "Rejected document" and call the `channel` function with the argument "legacy".
+
+=== Import Filter:
+[source,javascript]
+----
+function(doc) {
+    return doc.type == "mobile";
+}
+----
+When importing documents into collection1, there's an import filter that checks the `type` field value to determine if a document should be imported.
```
Comment (torcolvin) [inline modules/ROOT/pages/scopes-and-collections-config.adoc:140]: link to https://docs.couchbase.com/sync-gateway/current/import-filter.html ?

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 15]
```diff
@@ -0,0 +1,191 @@
...
+
+=== Sync Function
+[source,javascript]
+----
+function(doc, oldDoc, meta) {
+    channel("collection1");
+}
+----
+
+Whenever a document is synced or updated in `collection2`, its sync function is triggered.
```
Comment (torcolvin) [inline modules/ROOT/pages/scopes-and-collections-config.adoc:157]: One point I'm not sure how to address is that the sync function happens any time a document is changed, and sync doesn't have meaning, these are the times it happens:

- a document is updating independently in an a bucket on CBS by a process that is not sync gateway. If this happens and `auto_import` is true, the `import_filter` will decide whether the document is imported. If the document is imported the sync function will be run.
- a document is replicated from a couchbase lite client
- a document is created/modified through the sync gateway REST API

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 16]
```diff
@@ -0,0 +1,191 @@
+= Scopes and Collections Configuration for Sync Gateway
+:page-aliases: learn/scopes-and-collections-config.adoc
+ifdef::show_edition[:page-edition: {release}]
+ifdef::prerelease[:page-status: {prerelease}]
+:page-role:
+:page-content: conceptual
+:description: pass:q[How to configure Scopes and Collections for Sync Gateway - Examples with descriptions.]
+
+For a successful configuration of your Couchbase Sync Gateway Scopes and Collections, use this helpful example.
```
Comment (bfavini) [inline modules/ROOT/pages/scopes-and-collections-config.adoc]: This section explains how to configure Scopes and Collections for Sync Gateway. It describes configuration options and provides example configuration code.

**Understanding buckets, scopes, and collections**
Each database can contain multiple buckets. Each bucket can hold multiple scopes, and each scope can contain multiple collections.

When you create a bucket, a bucket._default._default element is inherently included. (see the documentation on [Default Scopes and Collections](https://github.com/couchbase/docs-sync-gateway/blob/be0913dac8a8f27a2f06554fd759f3ba9448a8a8/modules/ROOT/pages/server/learn/data/scopes-and-collections.adoc#default-scope-and-collection) in Couchbase Server.) Couchbase creates the default bucket to ensure backward compatibility.

Note: Only Couchbase Server (CBS) allows using the word 'default' in naming conventions. As a result,you may see the term _default.default when creating a new bucket.

File: modules/ROOT/pages/scopes-and-collections-config.adoc (hunk 9 above)
Comment (bfavini) [inline modules/ROOT/pages/scopes-and-collections-config.adoc]: Scopes are not required...

this note seems a little out of place. Move it after the bulleted list?

**Review by torcolvin — COMMENTED** [review]
Take or leave some of my comments.

I think the formatting is wrong since most of this should be italicized. I tried to follow the conventions of how https://docs.couchbase.com/server/current/learn/data/scopes-and-collections.html uses `_default` as a code literal. I'm not sure my suggestions are correct, I did not look at the rendering.

I think referring to what a sync function or import function looks like might be worthwhile to send to pages that cover them in detail.

I also wouldn't take any of my wording literally - it's off the cuff words to try to explain it to you.

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 17]
```diff
@@ -0,0 +1,191 @@
...
+Gain maximum flexibility in organizing and accessing your data by customizing it to suit your needs.
+
+[NOTE]
+====
+Scopes is not required for upgrading to Sync Gateway 3.1. 
+If it is not specified in the configuration, the previous versions are still compatible with the new version. Therefore, the provided examples are not necessary for upgrading to the latest version of Sync Gateway.
+====
+
+* *default scope/custom collection* +
+Create a custom collection with the default scope can lead to efficient data organization and quicker access control management.
```
Comment (bfavini) [inline modules/ROOT/pages/scopes-and-collections-config.adoc:32]: Create a custom collection with the default scope for efficient data organization and quicker access control managment.

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 18]
```diff
@@ -0,0 +1,191 @@
...
+
+* *default scope/default collection* +
+Work within the default Scopes and Collections. This makes managing data easy and simple without creating extra scopes or collections.
```
Comment (bfavini) [inline modules/ROOT/pages/scopes-and-collections-config.adoc:35]: Work with the default scopes and collections to make managing data easy without creating extra scopes or collections.

File: modules/ROOT/pages/scopes-and-collections-config.adoc (hunk 10 above)
Comment (bfavini) [inline modules/ROOT/pages/scopes-and-collections-config.adoc:39]: When you import a dataset...

File: modules/ROOT/pages/scopes-and-collections-config.adoc [hunk 19]
```diff
@@ -0,0 +1,191 @@
...
+      }
+    }
+  },
+  "num_index_replicas": 0
+}
+----
+
+== Collections Configuration Description
+
+Here is the description for the custom `Scope` with 2 custom `Collections` configuration.
```
Comment (bfavini) [inline modules/ROOT/pages/scopes-and-collections-config.adoc:90]: (Avoid starting a section with Here is, here are)

This section describes the custom scope

File: modules/ROOT/pages/scopes-and-collections-config.adoc (hunk 14 above)
Comment (bfavini) [inline modules/ROOT/pages/scopes-and-collections-config.adoc:140]: When importing documents into collection 1, an import filter checks the 'type' field...

**Review by bfavini — APPROVED** [review]
I approve this PR with these changes.

**Review by bfavini — APPROVED** [review]
I approve this PR with these changes.

File: modules/ROOT/pages/scopes-and-collections-config.adoc (hunk 10 above)
Comment (franrob-projects) [reply modules/ROOT/pages/scopes-and-collections-config.adoc:39]: Changed to:

When you import a dataset created in previous versions of Couchbase Server, the documents will automatically be saved in the `_default` scope and `_default` collection.

File: modules/ROOT/pages/scopes-and-collections-config.adoc (hunk 15 above)
Comment (franrob-projects) [reply modules/ROOT/pages/scopes-and-collections-config.adoc:157]: Added(as a note):

The sync function occurs every time a document is modified, regardless of whether or not it has any significance. Here are the examples when it happens:

A document within a CBS bucket is updated by a process other than Sync Gateway. If the auto_import feature is on, the import_filter will decide if the document is eligible for import. If it passes, the sync function will activate.

A document has been copied from a Couchbase Lite client.

A document can be created or modified by using the Sync Gateway REST API.

File: modules/ROOT/pages/scopes-and-collections-config.adoc (hunk 13 above)
Comment (franrob-projects) [reply modules/ROOT/pages/scopes-and-collections-config.adoc:129]: added..

File: modules/ROOT/pages/scopes-and-collections-config.adoc (hunk 14 above)
Comment (franrob-projects) [reply modules/ROOT/pages/scopes-and-collections-config.adoc:140]: Added..
